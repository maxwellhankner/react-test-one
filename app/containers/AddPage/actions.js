import {
  CHANGE_WORD,
  ADD_WORD,
  ADD_WORD_SUCCESS,
  ADD_WORD_ERROR,
} from './constants';

export function changeWord(word) {
  return {
    type: CHANGE_WORD,
    word,
  };
}

export function addWord(word) {
  return {
    type: ADD_WORD,
    word,
  };
}

export function wordAdded() {
  return {
    type: ADD_WORD_SUCCESS,
  };
}

export function wordLoadingError(error) {
  return {
    type: ADD_WORD_ERROR,
    error,
  };
}
