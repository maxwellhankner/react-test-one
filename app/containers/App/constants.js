export const LOAD_DATA = 'boilerplate/App/LOAD_DATA';
export const LOAD_DATA_SUCCESS = 'boilerplate/App/LOAD_DATA_SUCCESS';
export const LOAD_DATA_ERROR = 'boilerplate/App/LOAD_DATA_ERROR';
