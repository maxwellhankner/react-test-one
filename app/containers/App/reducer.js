import produce from 'immer';
import { LOAD_DATA_SUCCESS, LOAD_DATA, LOAD_DATA_ERROR } from './constants';

export const initialState = {
  loading: false,
  error: false,
  data: false,
};

/* eslint-disable default-case, no-param-reassign */
const appReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case LOAD_DATA:
        draft.loading = true;
        draft.error = false;
        draft.data = false;
        break;

      case LOAD_DATA_SUCCESS:
        draft.data = action.data;
        draft.loading = false;
        break;

      case LOAD_DATA_ERROR:
        draft.error = action.error;
        draft.loading = false;
        break;
    }
  });

export default appReducer;
