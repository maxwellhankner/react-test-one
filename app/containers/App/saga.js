import { put, call, takeLatest } from 'redux-saga/effects';
import { LOAD_DATA } from './constants';
import { dataLoaded, dataLoadingError } from './actions';
import request from '../../utils/request';

export function* getData() {
  try {
    const requestURL = '/api/data';
    const data = yield call(request, requestURL);

    yield put(dataLoaded(data));
  } catch (err) {
    yield put(dataLoadingError(err));
  }
}

export default function* rootSaga() {
  yield takeLatest(LOAD_DATA, getData);
}
