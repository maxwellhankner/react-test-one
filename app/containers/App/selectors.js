import { createSelector } from 'reselect';
import { initialState } from './reducer';

const selectGlobal = state => state.global || initialState;

const selectRouter = state => state.router;

const makeSelectData = () =>
  createSelector(
    selectGlobal,
    globalState => globalState.data,
  );

const makeSelectLoading = () =>
  createSelector(
    selectGlobal,
    globalState => globalState.loading,
  );

const makeSelectError = () =>
  createSelector(
    selectGlobal,
    globalState => globalState.error,
  );

const makeSelectLocation = () =>
  createSelector(
    selectRouter,
    routerState => routerState.location,
  );

export {
  selectGlobal,
  makeSelectData,
  makeSelectLoading,
  makeSelectError,
  makeSelectLocation,
};
