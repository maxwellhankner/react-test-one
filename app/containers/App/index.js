import React from 'react';
import { Helmet } from 'react-helmet';
import styled from 'styled-components';
import { Switch, Route } from 'react-router-dom';

import MainPage from 'containers/MainPage/Loadable';
import AddPage from 'containers/AddPage/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';
import { useInjectSaga } from 'utils/injectSaga';
import saga from './saga';

import GlobalStyle from '../../global-styles';

const AppWrapper = styled.div`
  max-width: calc(768px + 16px * 2);
  margin: 0 auto;
  display: flex;
  min-height: 100%;
  padding: 0 16px;
  flex-direction: column;
`;

export default function App() {
  useInjectSaga({ key: 'App', saga });

  return (
    <AppWrapper>
      <Helmet titleTemplate="Word List" defaultTitle="Word List">
        <meta name="description" content="Create a list of words" />
      </Helmet>
      <Switch>
        <Route exact path="/" component={MainPage} />
        <Route exact path="/add" component={AddPage} />
        <Route path="" component={NotFoundPage} />
      </Switch>
      <GlobalStyle />
    </AppWrapper>
  );
}
