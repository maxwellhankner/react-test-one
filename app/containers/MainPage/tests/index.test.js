import React from 'react';
import { ConnectedRouter } from 'connected-react-router';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import history from 'utils/history';

import { MainPage, mapDispatchToProps } from '../index';
import configureStore from '../../../configureStore';
import { loadData } from '../../App/actions';

describe('<MainPage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should fetch the words on mount', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <ConnectedRouter history={history}>
            <MainPage loadMyData={submitSpy} />
          </ConnectedRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    describe('loadMyData', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.loadMyData).toBeDefined();
      });

      it('should dispatch loadData when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.loadMyData();
        expect(dispatch).toHaveBeenCalledWith(loadData());
      });
    });
  });
});
