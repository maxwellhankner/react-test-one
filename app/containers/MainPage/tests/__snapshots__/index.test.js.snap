// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> should render and match the snapshot 1`] = `
<div>
  <h2>
    <span>
      List of very important words
    </span>
  </h2>
  <div
    class="Wrapper-sc-5dsnpv-0 dQithP"
  >
    <a
      class="StyledLink-sc-7lpfxd-0 eXvIyo"
      href="/add"
      primary="true"
    >
      add a word
    </a>
  </div>
</div>
`;
